File: Array and Lists/remove_duplicates_from_sorted_array.py
# Link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/

# Approach: Keep two pointers one iterating array and one that only increments when new elements found (current_element > previous_array)

class Solution(object):
    def removeDuplicates(self, nums):
        real_prog = 1
        
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                nums[real_prog] = nums[i]
                real_prog += 1
        return real_prog 

File: Array and Lists/remove_duplicates_from_sorted_array.cc
// Link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution 
{
    public:
    int removeDuplicates(vector<int>& nums) 
    {
        int counter = 1;
        
        if(nums.size() == 0)
        {
            return 0;
        }
        
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i] != nums[i-1])
            {
                nums[counter] = nums[i];
                counter++;
            }
        }
        return counter;
    }
};